rust/protocol/src/storage/traits.rs

```diff
@@ -20,11 +20,27 @@ pub enum Direction {
     Receiving,
 }
 
+/// A locally-generated random number used to construct the initial value of a message chain.
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
+pub struct RegistrationId(pub u32);
```

Comment:
These should not be `Ord`, as they are generally randomly generated.

For bonus points later we should actually make this a u16, because that's what the server expects, but that'll mess up all the bridges, so let's leave it for this PR.

Reply:
Fixed! Before reading this comment I was actually able to address the bridge part anyway by producing `unsafe_*` extractor functions in 89a79bb.